features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -135,13 +136,21 @@ static bool convert_mbed_addr_to_lwip(ip_addr_t *out, const nsapi_addr_t *in)
     }
 #if !LWIP_IPV6
     /* For symmetry with above, accept IPv6 :: as a general null */
-    if (in->version == NSAPI_IPv6 && all_zeros(in->bytes, 16)) {
+    if (in->version == NSAPI_UNSPEC ||
+        (in->version == NSAPI_IPv6 && all_zeros(in->bytes, 16))) {
         ip_addr_set_zero_ip4(out);
         return true;
     }
 #endif
 #endif
 
+#if LWIP_IPV4 && LWIP_IPV6
```

Comment:
Not sure about this flopping towards IPV4. Could it at least follow the IPv4/IPv6 preference? Ideally we'd convert to lwip's own IPADDR_TYPE_ANY, but that isn't fully supported in their ip_addr_t either - it's used internally, but I don't believe it's valid for external users.

Jesus, this ifdeffing is confusing. Not sure we can do better though.